src/hypothesis/strategies.py

```diff
@@ -836,31 +836,85 @@ def dm_func(denom):
-
-    return (
-        floats_strategy.map(float_to_decimal) |
-        fractions_strategy.map(lambda f: Decimal(f.numerator) / f.denominator)
-    )
+    # Convert min_value and max_value to Decimal values, and validate args
+    check_valid_integer(places)
+    if places is not None and places < 0:
+        raise InvalidArgument('places=%r may not be negative' % places)
+
+    if min_value is not None:
+        min_value = Decimal(min_value)
+        if min_value.is_infinite() and min_value < 0:
+            if not (allow_infinity or allow_infinity is None):
+                raise InvalidArgument('allow_infinity=%r, but min_value=%r'
+                                      % (allow_infinity, min_value))
+            min_value = None
+        elif not min_value.is_finite():
+            # This could be positive infinity, quiet NaN, or signalling NaN
+            raise InvalidArgument(u'Invalid min_value=%r' % min_value)
+    if max_value is not None:
+        max_value = Decimal(max_value)
+        if max_value.is_infinite() and max_value > 0:
+            if not (allow_infinity or allow_infinity is None):
+                raise InvalidArgument('allow_infinity=%r, but max_value=%r'
+                                      % (allow_infinity, max_value))
+            max_value = None
+        elif not max_value.is_finite():
+            raise InvalidArgument(u'Invalid max_value=%r' % max_value)
+    check_valid_bound(min_value, 'min_value')
+    check_valid_bound(max_value, 'max_value')
+    check_valid_interval(min_value, max_value, 'min_value', 'max_value')
+
+    if allow_infinity and (None not in (min_value, max_value)):
+        raise InvalidArgument('Cannot allow infinity between finite bounds')
+    # Set up a strategy for finite decimals
+    if places is not None:
+        # Fixed-point decimals are basically integers with a scale factor
+
+        def try_quantize(d):
+            try:
+                return d.quantize(factor)
+            except InvalidOperation:  # pragma: no cover
+                return None
+        factor = Decimal(10) ** -places
+        max_num = max_value / factor if max_value is not None else None
+        min_num = min_value / factor if min_value is not None else None
+        strat = integers(min_value=min_num, max_value=max_num)\
+            .map(lambda d: try_quantize(d * factor))\
+            .filter(lambda d: d is not None)
+    else:
+        # Otherwise, they're like fractions featuring a power of ten
+        strat = fractions(
+            min_value=min_value, max_value=max_value
+        ).map(lambda f: Decimal(f.numerator) / f.denominator)
+    # Compose with sampled_from for infinities and NaNs as appropriate
+    special = []
+    if allow_nan or (allow_nan is None and (None in (min_value, max_value))):
+        special.extend(map(Decimal, ('NaN', '-NaN', 'sNaN', '-sNaN')))
+    if allow_infinity or (allow_infinity is max_value is None):
```

Comment:
This is because of [Python's slightly silly comparison chaining rules](https://docs.python.org/2/reference/expressions.html#comparisons). `x is y is z` means something completely different from `(x is y) is z` or `x is (y is z)`.

It's a supposedly natural generalisation of the thing that makes `x <= y <= z` work.